clients/drcachesim/tracer/output.cpp

```diff
@@ -923,6 +923,35 @@ process_buffer_for_physaddr(void *drcontext, per_thread_t *data, size_t header_s
     return skip;
 }
 
+offline_entry_t *
+find_unfiltered_record(byte *start, byte *end)
+{
+    // The end variable points to the next writable location.
+    offline_entry_t *last = (offline_entry_t *)(end - sizeof(offline_entry_t));
+
+    int num_memrefs = 0;
+
+    for (offline_entry_t *entry = last; entry >= (offline_entry_t *)start; entry--) {
+        if (entry->pc.type == OFFLINE_TYPE_PC) {
+            NOTIFY(4, "PC: instr count = %d, num_memrefs = %d\n", entry->pc.instr_count,
+                   num_memrefs);
+            if ((entry->pc.instr_count == 1 && num_memrefs > 0) ||
+                entry->pc.instr_count > 1) {
```

Comment:
I don't know if a PC entry with 1 instruction is unfiltered or filtered (unless it has memrefs), so I am erring towards marking it as filtered. I have added some comments in the code to mention this.

Reply:
Please explain in the comment *why* we can't tell.  Is it because the mode change could be after the drbbdup dispatch reads the mode and before the end of block check?  In which case we'd have a filtered entry.  OK.  So we're living with this case where we might include an unfiltered record but present it as filtered.

Reply:
I updated the comment to address this.